crates/kernel_cmdline/src/bytes.rs

```diff
@@ -678,6 +708,49 @@ mod tests {
         assert_eq!(rd_args[3], param("rd.qux=c"));
     }
 
+    #[test]
+    fn test_add() {
+        let mut kargs = Cmdline::from(b"console=tty0 console=ttyS1");
+
+        // add new parameter with duplicate key but different value
+        assert!(matches!(
+            kargs.add(&param("console=ttyS2")),
+            Action::Added
+        ));
+        let mut iter = kargs.iter();
+        assert_eq!(iter.next(), Some(param("console=tty0")));
```

Comment:
Minor but I think we could use https://docs.rs/itertools/latest/itertools/fn.equal.html

Reply:
Yep totally fine, we can do that kind of stuff as async followup